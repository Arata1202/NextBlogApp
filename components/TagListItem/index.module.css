.tag {
  font-size: 1rem;
  border-radius: var(--border-radius);
  font-weight: bold;
  white-space: nowrap;
}
