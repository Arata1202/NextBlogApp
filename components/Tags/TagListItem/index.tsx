import React from 'react';
import { HashtagIcon } from '@heroicons/react/24/outline';
import { Tag } from '@/libs/microcms';
import styles from './index.module.css';

type Props = {
  tag: Tag;
  hasLink?: boolean;
};

export default function TagListItem({ tag, hasLink = true }: Props) {
  if (hasLink) {
    return (
      <>
        <a href={`/tag/${tag.id}`}>
          <HashtagIcon className="h-5 w-5 mt-1" />
          <div className={styles.tag}>{tag.name}</div>
        </a>
      </>
    );
  }
  return (
    <>
      <div className={styles.tag}>{tag.name}</div>
    </>
  );
}
