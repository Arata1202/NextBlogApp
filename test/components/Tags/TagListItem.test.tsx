import { render } from '@testing-library/react';
import TagListItem from '@/components/Tags/TagListItem';
import type { Tag } from '@/libs/microcms';

describe('TagListItem', () => {
  test('TagListItem)', () => {
    const tag: Tag = { name: 'example', id: '1', createdAt: '', updatedAt: '' };
    const { asFragment } = render(<TagListItem tag={tag} hasLink={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
