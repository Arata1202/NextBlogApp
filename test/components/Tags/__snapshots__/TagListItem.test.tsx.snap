// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagListItem スナップショット(TagListItem) 1`] = `
<DocumentFragment>
  <a
    class="tag"
    href="/tag/1"
  >
    <svg
      aria-hidden="true"
      class="h-5 w-5 mt-1"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.25 8.25h15m-16.5 7.5h15m-1.8-13.5-3.9 19.5m-2.1-19.5-3.9 19.5"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <p
      class="tag"
    >
      example
    </p>
  </a>
</DocumentFragment>
`;
